src/controllers/submissionController.ts

```diff
@@ -110,24 +115,32 @@ export const submitProject = async (
   req: Request,
   res: Response
 ): Promise<void> => {
+  const submissionRepo = getManager().getRepository(Submission);
+  const challengeRepo = getManager().getRepository(Challenge);
+  const teamRepo = getManager().getRepository(Team);
   const token = req.header('Authorization')?.split(' ')[1];
-  if (!token) {
-    res.sendStatus(401); // User was not properly authenticated...
-    return;
+  if(!token) {
+      res.sendStatus(401);
+      return;
   }
 
-  const uuid = auth.getUserFromToken(token);
-  if (!uuid) {
-    res.sendStatus(401); // User was not properly authenticated...
-    return;
+  const user = await auth.getUserObjectFromToken(token, ['team']);
+  if(!user) {
+      res.sendStatus(401);
+      return;
   }
 
-  // TODO: Check if user is in team, and that team has not already submitted project
+  if(!user.team) {
+    res.status(400).send('User is not on a team');
+    return;
+  }
```

Comment:
How are we handling teams that have already submitted? Are we just going with replacing the existing submission if they submit again?

Reply:
I use Object.assign, so basically, any provided field will get overwritten if they submit again. So essentially, it edits their submission.